# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteImageHolder.h
# Commit: afef5857aea6
# Full Hash: afef5857aea6cbbd3e9296ce39c004812967d095
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-08 04:52:24
# Regressor Bug: 1787561
# File Overlap Count: 1
# Description:
#   Bug 1787561 Part 1: Make GPUVideoImage track color depth, and make RemoteImageHolder pass it through. r=media-playback-reviewers,gfx-reviewers,alwu,aosmond
#   
#   This ensures that the Image that the client receives has the color depth
#   information that was sent to the GPU, even though the client can't query
#   the GPU directly.
# ==============================================================================

diff -r 1b2dc91d3c38 -r afef5857aea6 dom/media/ipc/RemoteImageHolder.h
--- a/dom/media/ipc/RemoteImageHolder.h	Wed Sep 07 18:23:24 2022 +0000
+++ b/dom/media/ipc/RemoteImageHolder.h	Wed Sep 07 18:31:45 2022 +0000
@@ -28,6 +28,7 @@
   RemoteImageHolder(layers::IGPUVideoSurfaceManager* aManager,
                     layers::VideoBridgeSource aSource,
                     const gfx::IntSize& aSize,
+                    const gfx::ColorDepth& aColorDepth,
                     const layers::SurfaceDescriptor& aSD);
   RemoteImageHolder(RemoteImageHolder&& aOther);
   // Ensure we never copy this object.
@@ -47,6 +48,7 @@
   // We need a default for the default constructor, never used in practice.
   layers::VideoBridgeSource mSource = layers::VideoBridgeSource::GpuProcess;
   gfx::IntSize mSize;
+  gfx::ColorDepth mColorDepth;
   Maybe<layers::SurfaceDescriptor> mSD;
   RefPtr<layers::IGPUVideoSurfaceManager> mManager;
 };